samples/nrf9160/gps/src/main.c

```diff
@@ -126,65 +130,92 @@ static int init_app(void)
 					       NRF_GNSS_NMEA_GGA_MASK |
 					       NRF_GNSS_NMEA_RMC_MASK;
 
-	if (setup_modem() != 0) {
-		printk("Failed to initialize modem\n");
-		return -1;
-	}
+	if (ctrl == GNSS_INIT_AND_START) {
```

Comment:
Overall this entire function looks like a good candidate for a `switch (ctrl) {` block, with fallthrough from `GNSS_INIT_AND_START` into `GNSS_RESTART`. Consider it just as a suggestion, up to you to change.

Reply:
noted, no change